profit/fit.py

```diff
@@ -16,43 +13,37 @@ def get_surrogate(sur):
 
 class ActiveLearning:
 
-    def __init__(self, config):
-        from profit.run import LocalCommand
+    def __init__(self, config, runner):
         self.config = config
-        x = load(self.config['files']['input'])
-        self.x = np.hstack([x[key] for key in x.dtype.names])
-        self.y = None
-        self.run = LocalCommand(config['run']['cmd'], config['run']['ntask'],
-                                run_dir=config['run_dir'], base_dir=config['base_dir'])
+
+        self.runner = runner
+
         self.sur = get_surrogate(config['fit']['surrogate'])
 
-    def update_run(self, krun, al_value):
-        #y_single = np.array([[self.f(*self.x[krun])]])
+        self.x = self.runner.flat_input_data
```

Comment:
`self.x` has a shape of `(64, 1)` instead of `(ntrain, 1)` per default. The first `ntrain` values are filled correctly with `nan` and the rest with zeros. It still works, but is this intentional and efficient?

Reply:
64 is defined as the maximum size of the `memmap`. But you’re right, it’s probably better to initialize with `ntrain` as we never need more runs than `ntrain`.